### ISSUE TITLE
[REQ] LostFilm.tv

### ISSUE THREAD

**Issue by krin-san** [Jackett/Jackett#1922]
Requesting support for Russian Semi-Private Tracker LostFilm.tv

Tracker Name: LostFilm.tv
Tracker Content: Huge amount of TV Series in Russian language.
Tracker URL : http://www.lostfilm.tv/
Tracker Language : Russian
Privacy: Free registration. Required to fetch torrent files. Not required for search functionality.

It has RSS feed available on http://retre.org/rssdd.xml with Cookies Auth (grants with a free regisration on LostFilm.tv) but Sonarr can't use Search with RSS indexer, incorrectly matches language and quality and (if everything previous were solved using Restrictions) can't download torrent file by link in RSS feed.

Thank you.

**Comment by krin-san** [comment]
I just started looking at the [Definition-format](https://github.com/Jackett/Jackett/wiki/Definition-format) and it looks quite easy to configure on my own.

The trickiest part is **login** and **search**. It looks like Lostfilm has it's own series database where series_id not matches to any public database (like thetvdb). Also it use http://retre.org/ with some magic parameters to show a page with .torrent file download links which means we have to make few requests in order to get a link.

### Investigation results

https://www.lostfilm.tv/series/Breaking_Bad/seasons
Page sources contains the following line:
```
var serial_id = '119';
```
As I saw html sources is the only place to grab **series_id** which will be needed in future.

Clicking on download (_red disk_) button which is available noly for logged in users for Season 03 Episode 16 leads us to the next page:
https://www.lostfilm.tv/v_search.php?c=119&s=5&e=16

Here we see both `series_id=119`, `season=5` and `episode=16`. For the whole season download links `episode=999`. But (if we was logged in) gets immediately redirected to the next url:
http://retre.org/v3/index.php?c=119&s=5&e=16&u=2271477&h=4eaa5debaf6bf34d622bb69ab77f3b64&n=1

New parameter `h` is a Hash which magically calculates somewhere on lostfilm server-side. It's another reason why we can't directly search for the links using http://retre.org

And here on the page we could see direct links to the torrent files like this one:
http://tracktor.in/td.php?s=3esrcccUQ8TP%2BXDpSLc%2FIao3X6gOphZSbOTkd2v%2FM9Bcei46SBGk6fUIrV0Dp6J7xCT6kujRnjr0JMkR9VbO7Xl7KT9xANgOJgL%2FD8S%2BA8MHe%2FEbup9bxLgXch2zJspg

### TL;DR

Any ideas on how to implement all that things? =]

**Comment by krin-san** [comment]
Ok, so I wrote a new indexer LostFilm, everything compiles but I can't figure out how to debug this thing. Running JackettService produce error saying "impossible to start service from command line or debugger ...".

How do you usually debugs new indexers while implementing them?

**Comment by krin-san** [comment]
Some results achieved! Browsing with empty search string & without season/episode definition left.
![image](https://user-images.githubusercontent.com/2318002/31565251-c76c8ce6-b06e-11e7-8258-07247096e1e9.png)
Still not sure what to do with *PublishDate* - there are no such info on retre.org webpage.

**Comment by krin-san** [comment]
Just finished implementing a 500+ lines indexer and noticed fun fact: LostFilm replaces episode releases with a single season releases even though you're trying to download only single episode from it.
![image](https://user-images.githubusercontent.com/2318002/31633016-a76d26b6-b2c7-11e7-9ea5-9b1a56884579.png)
![image](https://user-images.githubusercontent.com/2318002/31633047-bcdeb7da-b2c7-11e7-9648-abfd887ba06a.png)
So this indexer could work only with newest releases and Extras episodes.

**Comment by garfield69** [comment]
possible shutdown
http://blocked.mts.ru/?host=www.lostfilm.tv

**Comment by janot** [comment]
It's only ISP block page

**Comment by garfield69** [comment]
not by my ISP, so presumably everyone is blocked?

**Comment by garfield69** [comment]
no, i see now. if I use a Canadian vpn then i can get in.

**Comment by ilike2burnthing** [comment]
https://www.lostfilm.run/ - SSL handshake failed - Error code 525
https://www.lostfilmtv.site/ - SSL handshake failed - Error code 525
https://www.lostfilm.uno/ - Origin is unreachable - Error code 523

Working:
https://www.lostfilm.tv/
https://www.lostfilm.win/
https://www.lostfilm.tw/
https://www.lostfilmtv2.site/

Give it a week and then get rid of them if they're still down.

**Comment by garfield69** [comment]
I've tested the indexer against the 3 you found errors, and they are all working for me.
In addition found another working domain https://www.lostfilmtv5.site/

**Comment by ilike2burnthing** [comment]
@garfield69 & @mynameisbogdan `webclient.AddTrustedCertificate(new Uri(SiteLink).Host` only works for the primary sitelink, otherwise it seems to be ignored and results in invalid cert errors (see *.tw above). This can cause the indexer  config panel to be inaccessible unless deleted and re-added.

However, `certificates` work fine in YAML indexers, no matter if they're for the primary sitelink or not.

It's an easy enough workaround here, make *.tw the primary sitelink, but if an indexer needs multiple exclusions, or the exclusion is for a proxy/mirror that we'd prefer not to make primary, this obviously won't work.

Any fix?

**Comment by garfield69** [comment]
The routine that cardigann calls for yaml indexers is the same webclient.AddTrustedCertificate routine that the lostfilm c# indexer calls.
I think that the underlying issue here is that the *.tw link returns ERR_CONNECTION_RESET which prevents the webclient accessing the site and prevents SSL from checking the CA.
If any of the yaml indexers had links that returned the same error I would expect that cardigann would also fail in the same way that the lostfilm c# indexer does.

I believe there is no fix, other than to move that *.tw link to the legacy list.

**Comment by garfield69** [comment]
also note that that CA bypass was added for *.tw when *.tw was online but had an invalid CA.
All the other links for lostfilm have valid CA of their own so that bypass does not apply to them.

**Comment by ilike2burnthing** [comment]
https://www.lostfilm.life/ - Web server is returning an unknown error Error code 520
Other domains are working fine. If it's not back up in a week or so we can remove.

**Comment by garfield69** [comment]
all domains online

**Comment by garfield69** [comment]
https://www.lostfilm.tv/ > cloudflare Connection timed out Error code 522
https://www.lostfilmtv.site/ > Please check back later Error 1027  This website has been temporarily rate limited
https://www.lostfilmtv5.site/ > HTTP ERROR 522
https://www.lostfilmtv2.site/ > cloudflare Connection timed out Error code 522
https://www.lostfilmtv3.site/ > cloudflare Invalid SSL certificate Error code 526
https://www.lostfilm.today/ > cloudflare Connection timed out Error code 522
https://www.lostfilm.download/ > cloudflare Connection timed out Error code 522
https://www.lostfilm.life/ > cloudflare Connection timed out Error code 522
https://www.lostfilm.uno/ > ERR_CONNECTION_TIMED_OUT
https://www.lostfilm.win/ > DNS_PROBE_FINISHED_NXDOMAIN
https://www.lostfilm.tw/ net::ERR_CERT_DATE_INVALID

**Comment by garfield69** [comment]
https://www.lostfilm.tv/ online
https://www.lostfilmtv.site/ > online
https://www.lostfilm.uno/ online
https://www.lostfilm.tw/ online
https://www.lostfilm.download/ online
https://www.lostfilm.life/ online

remaining alternate domains to be culled if not resolved in 30 days from now
https://www.lostfilmtv5.site/ > cloudflare Invalid SSL certificate Error code 526
https://www.lostfilmtv2.site/ > cloudflare Invalid SSL certificate Error code 526
https://www.lostfilmtv3.site/ > HTTP ERROR 526

**Comment by ilike2burnthing** [comment]
all up

**Comment by garfield69** [comment]
https://www.lostfilm.tv/ > cloudflare A timeout occurred Error code 524
https://www.lostfilmtv.site/ > DNS_PROBE_FINISHED_NXDOMAIN
https://www.lostfilmtv5.site/ > cloudflare A timeout occurred Error code 524
https://www.lostfilmtv3.site/ > cloudflare A timeout occurred Error code 524
https://www.lostfilmtv2.site/ > cloudflare A timeout occurred Error code 524
https://www.lostfilm.today/ > cloudflare A timeout occurred Error code 524
https://www.lostfilm.download/ > cloudflare A timeout occurred Error code 524
https://www.lostfilm.life/ > cloudflare A timeout occurred Error code 524
https://www.lostfilm.uno/ > cloudflare A timeout occurred Error code 524
https://www.lostfilm.tw/ > invalid CA > I gave up waiting for it on firefox, chrome wont let me even try because of CA

**Comment by garfield69** [comment]
online

**Comment by garfield69** [comment]
There is a problem with the lostfilm sites when it calls ajaxik.php, which the site uses to login and logout.
This prevents users that need to login or logout (browser or Jackett indexer), because the site returns a 502 Bad Gateway
Only users that are already logged in can do searches.

**Comment by avt00** [comment]
> There is a problem with the lostfilm sites when it calls ajaxik.php, which the site uses to login and logout. This prevents users that need to login or logout (browser or Jackett indexer), because the site returns a 502 Bad Gateway Only users that are already logged in can do searches.

I've just logged in on the browser without any issues. Would it be possible to somehow pass the session cookie to Jackett?

**Comment by garfield69** [comment]
I'll see if switching the indexer to the cookie method works.

**Comment by garfield69** [comment]
Using cookies does not help, because the LostFilm C# indexer uses the API (ajaxik.php) to also do the searches, so back to the intermittent 502 Bad Gateway error :-(

**Comment by avt00** [comment]
> Using cookies does not help, because the LostFilm C# indexer uses the API (ajaxik.php) to also do the searches, so back to the intermittent 502 Bad Gateway error :-(

Thanks for trying the cookie method. I'll try to reach out to the tracker team and see if they can fix the file.

**Comment by garfield69** [comment]
the lostfilm indexer appears to be working for me again.

**Comment by garfield69** [comment]
https://github.com/Jackett/Jackett/issues/16074#issuecomment-3090681344

**Comment by garfield69** [comment]
0f88863e5d6334fc906898ad4ff4405b0b53638a
